website/docs/concepts/wasm-bindgen.md

```diff
@@ -0,0 +1,239 @@
+you can think of it as the `A` in the example above, this type is [`JsValue`](#jsvalue) which has
+it's own section
+below.
+
+_[extends section in The `wasm-bindgen` Guide](https://rustwasm.github.io/docs/wasm-bindgen/reference/attributes/on-js-imports/extends.html)_
+
+### [`JsValue`](https://rustwasm.github.io/wasm-bindgen/api/wasm_bindgen/struct.JsValue.html)
+
+This is a representation of an object owned by JavaScript, this is a root catch-all type for `wasm-bindgen`.
+Any other type that comes from `wasm-bindgen` is a `JsValue` and this is because JavaScript doesn't have
+a strong type system so any function that accepts a variable `x` doesn't define it's type so `x` can be
+an valid JavaScript value; hence `JsValue`. So when you are working with imported functions or types that
+accept a `JsValue` then any value is _technically_ valid.
+
+`JsValue` can be accepted by a function but that function may still only expect certain types and this
+can lead to panics - so when using raw `wasm-bindgen` APIs check the documentation of the JavaScript
+being imported whether an exception will be caused if that value is not a certain type.
+
+_[`JsValue` documentation](https://rustwasm.github.io/wasm-bindgen/api/wasm_bindgen/struct.JsValue.html)._
+
+### [`JsCast`](https://rustwasm.github.io/wasm-bindgen/api/wasm_bindgen/trait.JsCast.html)
+
+Rust has a strong type system and JavaScript...doesn't 😞 So in order for Rust to maintain these
+strong types but still be convenient the web assembly group came up with a pretty neat trait `JsCast`.
+It's job is to help you move from one JavaScript "type" to another, which sounds vague but it means
+that if you have one type which you know is really another then you can use the functions of `JsCast`
+to jump from one type to the other. It's a nice trait to get to know when working with `web-sys`,
+`wasm_bindgen`, `js-sys` - you'll notice lots of types will implement `JsCast` from those crates.
+
+`JsCast` provides both checked and unchecked methods of casting - so that at runtime if you are
+unsure what type a certain object is you can try to cast it which returns possible failure types like
+[`Option`](https://doc.rust-lang.org/std/option/enum.Option.html) and
+[`Result`](https://doc.rust-lang.org/std/result/enum.Result.html()).
+
+A common example of this in [`web-sys`](wasm-bindgen/web-sys) is when you are trying to get the
+target of an event, you might know what the target element is but the
+[`web_sys::Event`](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.Event.html) API will always return an [`Option<web_sys::EventTarget>`](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.Event.html#method.target)
+so you will need to cast it to the element type so you can call it's methods.
+
+```rust
+// yew exports some of web_sys
+use yew::web_sys::{Event, EventTarget, HtmlInputElement, HtmlSelectElement};
```

Comment:
Not sure if we should use the re-exports here since I'm actively trying to get rid of them :P
Especially beginners might not realize how to activate additional features (since Yew doesn't activate all of them) and Cargo seems to be trying to move  away from the transitive relations in cargo features as well.

Reply:
I wasn't sure about it to be honest and I ended up adding them when the example was integrating with Yew, because we do re-export them.  If you use an IDE, one that may or may not also like to switch between tabs and space indents on you 😞, auto-correct will often pull in the re-export of  `web-sys` so I think beginners will run into this anyway at least this way its documented to some degree. 

> Especially beginners might not realize how to activate additional features (since Yew doesn't activate all of them)

I did add a small note at the end about this.

> I'm actively trying to get rid of them :P

Should I make an issue / PR and just do it then? >:D Your answer here will probably change my argument above :)

Reply:
Unless you're very attached to the re-export, yeah, I vote to remove it.

Reply:
Updated this to import through `web_sys` directly ready for #2084 👀